### PR TITLE
Universal fixture and getting Ruff with the repo

### PR DESCRIPTION
**Pull request by ialarmedalien**
- Various small fixes suggested by Ruff -- mainly converting for loops into list comprehensions and `"blah blah {}".format("blah")` to f-strings.
- Create conftest file and add fixture to create the various clients and workspaces needed

### CODE REVIEW

File: lib/SetAPI/SetAPIImpl.py [hunk 1]
```diff
@@ -2148,7 +2148,7 @@ def sample_set_to_samples_info(self, ctx, params):
         #BEGIN sample_set_to_samples_info
         sample_search_utils = SamplesSearchUtils(ctx['token'], self.search_url)
         if not params.get('ref'):
-          raise ValueError(f"Argument 'ref' must be specified, 'ref' = '{params.get('ref')}' ")
+          raise ValueError(f"Argument 'ref' must be specified, 'ref' = '{params.get('ref')}'")
```
Comment (ialarmedalien) [inline lib/SetAPI/SetAPIImpl.py:2151]: trailing space

File: lib/SetAPI/sampleset/SampleSetInterface.py [hunk 2]
```diff
@@ -15,11 +15,11 @@ def __init__(self, workspace_client):
     def _ws_get_ref(self, ws_id, obj_id):
         if "/" in obj_id:
             return obj_id
-        else:
-            info = self.ws_client.get_object_info_new(
-                {"objects": [{"name": obj_id, "workspace": ws_id}]}
-            )[0]
-            return "{0}/{1}/{2}".format(info[6], info[0], info[4])
+
+        info = self.ws_client.get_object_info_new(
```
Comment (ialarmedalien) [inline lib/SetAPI/sampleset/SampleSetInterface.py:19]: else not needed after `return`
convert to f-string

File: test/AssemblySet_basic_test.py [hunk 3]
```diff
@@ -3,16 +3,16 @@
 import shutil
 import unittest
 from pprint import pprint
-
 from test import TEST_BASE_DIR
-from test.test_config import get_test_config
```
Comment (ialarmedalien) [inline test/AssemblySet_basic_test.py:8]: converted the `test_config` file into `conftest`, and renamed `get_test_config` just `test_config`

File: test/conftest.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    "KB_DEPLOYMENT_CONFIG",
+    os.path.join(TEST_BASE_DIR, "./deploy.cfg")
+)
+
+WS_NAME = "test_SetAPI_" + str(int(time.time() * 1000))
+TOKEN = os.environ.get("KB_AUTH_TOKEN", None)
+SDK_CALLBACK_URL = os.environ["SDK_CALLBACK_URL"]
+INFO_LENGTH = 11
+
+@pytest.fixture(scope="session")
```
Comment (briehl) [inline test/conftest.py:27]: Very nice!

File: test/ExpressionSet_basic_test.py [hunk 5]
```diff
@@ -67,31 +69,31 @@ def setUpClass(cls):
...
         # name, genome_ref, annotation_ref, alignment_ref, ws_name, ws_client
-        cls.expression_refs = list()
+        cls.expression_refs = []
         for idx, alignment_ref in enumerate(cls.alignment_refs):
             cls.expression_refs.append(
                 make_fake_expression(
                     os.environ["SDK_CALLBACK_URL"],
                     cls.dummy_path,
-                    "fake_expression_{}".format(idx),
+                    f"fake_expression_{idx!s}",
```
Comment (briehl) [inline test/ExpressionSet_basic_test.py:84]: Was the `!s` bit added by Ruff? It's not really necessary, but if autogenerated, it's fine.

**Review by briehl — APPROVED** [review]
Just a quick question, but looks great!

**Pull request merged**